keps/sig-instrumentation/3077-contextual-logging/README.md

```diff
@@ -0,0 +1,1000 @@
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+    - [Story 3](#story-3)
+    - [Story 4](#story-4)
+  - [Risks and Mitigations](#risks-and-mitigations)
+    - [Uninitialized logger](#uninitialized-logger)
+    - [Logging during initialization](#logging-during-initialization)
+    - [Performance overhead](#performance-overhead)
+    - [Transition](#transition)
+    - [Pitfalls during usage](#pitfalls-during-usage)
+      - [Logger in context and function not consistent](#logger-in-context-and-function-not-consistent)
+      - [Overwriting context not needed initially, forgotten later](#overwriting-context-not-needed-initially-forgotten-later)
+      - [Redundant key/value pairs](#redundant-keyvalue-pairs)
+      - [Modifying the same variable in a loop](#modifying-the-same-variable-in-a-loop)
+      - [Unused return values](#unused-return-values)
+- [Design Details](#design-details)
+  - [Text format](#text-format)
+  - [Default logger](#default-logger)
+    - [logging helper API](#logging-helper-api)
+    - [Logging in tests](#logging-in-tests)
+  - [<a href="https://github.com/kubernetes/klog/tree/main/hack/tools/logcheck">logcheck</a>](#logcheck)
+    - [Code examples](#code-examples)
+      - [klogr replacing klog completely, explicit logger parameter](#klogr-replacing-klog-completely-explicit-logger-parameter)
+      - [Unit testing](#unit-testing)
+      - [Injecting common names and values, logger passed through existing ctx parameter](#injecting-common-names-and-values-logger-passed-through-existing-ctx-parameter)
+      - [Resulting output](#resulting-output)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Alpha](#alpha)
+    - [Beta](#beta)
+    - [GA](#ga)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [Per-component logger](#per-component-logger)
+  - [Propagating a logger to init code](#propagating-a-logger-to-init-code)
+  - [Panic when DefaultLogger is called too early](#panic-when-defaultlogger-is-called-too-early)
+- [Infrastructure Needed](#infrastructure-needed)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
```

Comment:
We could already mark some of those points of the list so release team could add KEP to milestone.

Reply:
Will do.